cooker/bin/commands/start/request-listener.js

```diff
@@ -243,6 +245,21 @@ async function requestListener(req, res) {
     }
   }
 
+  if(config.OPTIONS.plugins) {
+    for (let i = 0; i < config.OPTIONS.plugins.length; i++) {
+      const plugin = config.OPTIONS.plugins[i];
+      
+      if(plugin.endPoint && typeof plugin.endPoint === 'function') {
+        const pluginResult = plugin.endPoint(req.url);
+        if(pluginResult && pluginResult.body) {
+          const contents = pluginResult.type === 'html' ? plugSocketIO(pluginResult.body) : pluginResult.body
+          _respond(res, { contentType: `text/${pluginResult.type}`, contents });
```

Comment:
If you have multiple plugins transforming the same type of content, you would be returning the response right after the first one.
[snowpack has a really nice plugin architecture](https://www.snowpack.dev/guides/plugins) which I recommend as inspiration.

Reply:
For now the plugins can only return a response to an endpoint, not transform anything. So if a plugin has an endpoint for '/about' it'll return the plugins result and not the generated about page. We could add a warning for those scenarios? If 2 plugins are listening to the same endpoint it'll return the first plugin with a response.

Reply:
Got it! Would mind elaborating with an example @ste-vg ?

Reply:
OK, lets say you want to build a localizations tool that lets you see and compare all the projects copy and translations on one page. With this setup it allows you to create a plugin that 'listens' for a page at '/locales/' and will grab all the relevant YAML files in the project and generate a report based on them. 

A very simple version of this would be:

```js
const LocalesTool = {
  endPoint: (url) => {
    if(url === '/locales/') { 
       // url matches so plugin will return something
      let html = someFunctionToFetchAllLocalesData();
      return { type: 'html', body: html }
    }

     // Url doesn't match this plugin so return null
     return null;
  }
}

module.exports = LocalesTool
```

You then add this plugin to the project in the sweet-potato.config.js like so:

```js
const LocalesTool = require('./LocalesPlugin.js');

module.exports = {
  sourceDirectory: 'src',
  plugins: [ LocalesTool ]
}
```

At the moment this stuff only works in `npm run start`. In a future update I'd like to add a build option to the plugins that'll allow them to create all it's pages during the build script (taking into account any env settings).

Reply:
Got it! Thanks @ste-vg 
That makes sense, maybe we change the name then, from `endPoint` to something more event driven like `onRouteChange` or something, what do you think?